### ISSUE TITLE
Cookies: `__Secure-`/`__Host-` prefixes

### ISSUE THREAD

**Issue by craigfrancis** [mozilla/http-observatory#63]
Now that `__Secure-`/`__Host-` prefixes are supported in Chrome 49:

https://www.chromestatus.com/feature/4952188392570880

And might be coming to Firefox:

https://wiki.mozilla.org/Security/Guidelines/Web_Security#Experimental_Directives

Can we get extra points for using them?

https://tools.ietf.org/html/draft-west-cookie-prefixes-05

**Comment by marumari** [comment]
Yep, they were on the docket, but there had been so many changes to the spec that I wanted to wait until it was a bit more set in stone.  But the eventual plan is to give +5 for using these prefixes.  :)

**Comment by craigfrancis** [comment]
Good point, the prefix was recently changed (Nov 2015) to use double underscores (rather than a dollar sign, which caused a couple of problems).

But as far as I can tell, this is the final version (and is implemented in Google Chrome).

Further discussion at:

https://twitter.com/tbroyer/status/670779401135616000

**Comment by marumari** [comment]
Note that our own guidelines talk about this:

https://wiki.mozilla.org/User:Apking/Web_Security_Guidelines#Cookies

Are you aware if anybody besides Chrome has implemented this yet?

**Comment by craigfrancis** [comment]
Not yet, but I have created feature requests for Safari and IE:

https://bugs.webkit.org/show_bug.cgi?id=158730

https://wpdev.uservoice.com/forums/257854-microsoft-edge-developer/suggestions/14807901-add-secure-and-host-cookie-prefix-suppor

**Comment by rugk** [comment]
Actually the prefix  is still ugly as the `-` [causes problems](https://serverfault.com/questions/788207/how-to-use-a-hypen-in-cookie-names-for-nginx).
So personally I hope the standard changes again...

**Comment by tobijdc** [comment]
:wave:

I have recently learned about cookie prefixes and their usefulness.
I was surprised observatory did not check them.
[It seems that Safari (at least recently) also supports these prefixes.](https://caniuse.com/mdn-http_headers_set-cookie_cookie_prefixes)

Is there interest for giving a small bonus score, if cookies additionally use these prefixes?
I am not a python developer, but I could give this a try over the Holidays, if there would be a willingness to merge this.

References:
[Mozilla Infosec Guidelines](https://infosec.mozilla.org/guidelines/web_security.html#directives-2), not marked experimental anymore [like in the Mozilla wiki](https://wiki.mozilla.org/Security/Guidelines/Web_Security#Experimental_Directives)
[MDM1](https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies#cookie_prefixes)
[MDM2](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie#cookie_prefixes)
